packages/flame/lib/src/effects/controllers/callback_controller.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:ui';
+import 'package:flame/effects.dart';
+
+/// The [callback] will be invoked once the effect's [progress] is reached.
+/// 1.0 to match `duration` or 0.0 to match when the `reverseDuration`
+/// has been reached.
+class CallbackController extends DurationEffectController {
+  CallbackController({required this.callback, required double progress})
+      : _progress = progress,
+        super(0.0);
+
+  final VoidCallback callback;
+  final double _progress;
+
+  @override
+  double get progress => _progress;
+
+  @override
+  bool get completed => true;
+
+  @override
+  double advance(double dt) {
+    callback();
+    return 0.00001;
```

Comment:
Shouldn't this (and recede) just return `dt`?

Reply:
0 I mean

Reply:
0 would mean that the controller still runs which blocks the effect. But returning dt seems to work fine too.

Reply:
I guess this is just to pass on the dt to the next controller in the chain.

Reply:
This should be `dt`, see the docs:

> /// If the controller is still running, the return value will be 0. If it
  /// already finished, then the return value will be the "leftover" part of
  /// the [dt]. That is, the amount of time [dt] that remains after the
  /// controller has finished. In all cases, the return value can be positive
  /// only when `completed == true`.

Reply:
It has been changed to return dt meanwhile.